### PR TITLE
docs: Add architecture diagram

### PR DESCRIPTION
**Pull request by surajssd**

### CODE REVIEW

File: Documentation/architecture.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+│   ├── common.bash
+│   ├── fedora.bash
+│   └── ubuntu.bash
+├── provision.tf
+├── terraform.tfvars.tmpl
+├── variables.tf
+└── vpc.tf
+```
+
+* `provider.tf` currently the only provider is AWS.
```
Comment (pothos) [inline Documentation/architecture.md]: `provider.tf` configures the provider which currently is only AWS.

File: Documentation/architecture.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Architecture
```
Comment (pothos) [inline Documentation/architecture.md]: Please link to this file from the README.

File: Documentation/architecture.md (hunk 1 above)
Comment (surajssd) [reply Documentation/architecture.md]: done!

File: Documentation/architecture.md (hunk 2 above)
Comment (surajssd) [reply Documentation/architecture.md]: done as well

**Pull request merged**